### ISSUE TITLE
[4.0] [RFC] Tinymce front-end css

### ISSUE THREAD

**Issue by Scrabble96** [joomla/joomla-cms#36584]
### Problem identified
Tinymce, which is shipped in Joomla core, uses the Cassiopeia css for all its styling on the front end. It is the same in J3, too, using Protostar CSS. This means if you write your own template you have to copy the relevant Joomla default site template CSS or write new CSS into your own template.

### Proposed solution
Use Joomla Core admin CSS to style Tinymce on the front end.

**Comment by dgrammatiko** [comment]
You can override the css per template

**Comment by Scrabble96** [comment]
> You can override the css per template

I have done that manually, as I said in the original post, but it means going back and forth from inspecting Cassiopeia layout and copying CSS, adding it to the new template CSS and then checking it in the new layout. A bit of a pain. There is a JCE template override for the pop-up, but that's no use if not using JCE.

Rather than using admin CSS, could the tinymce have its own CSS in the templates > system folder?
Or perhaps its own template override?

**Comment by dgrammatiko** [comment]
You can use this: https://wysiwyg.dgrammatiko.dev

It will give you way better results, actual WYSIWYG, from the core or any other 3rd PD editor

**Comment by Scrabble96** [comment]
Oops. I've just found the Joomla > Layouts > Tinymce but all it does is create an empty folder. I shall add a CSS file and test it now.

It would be more useful if it included the CSS in the override.

**Comment by dgrammatiko** [comment]
Do you want to style tinyMCE or the content inside the editor?

**Comment by brianteeman** [comment]
> Joomla > Layouts > Tinymce

That is not for css!!

See https://www.youtube.com/watch?v=OyZQ63Ahnt4&ab_channel=LearnJoomla4

**Comment by Scrabble96** [comment]
tinyMCE's own layout. This is because my template didn't include 'modal' CSS as I don't use modals anywhere, so the initial view is like this:
![tinymce-1](https://user-images.githubusercontent.com/36987494/148531266-0c58bba3-e074-4a5e-8fd4-ff47fb479df0.jpg)

**Comment by Scrabble96** [comment]
> > Joomla > Layouts > Tinymce
> 
> That is not for css!!
> 
> See https://www.youtube.com/watch?v=OyZQ63Ahnt4&ab_channel=LearnJoomla4

Yes, I watched that yesterday and saw it's for text snippets and templates, but thought it might work for CSS, too.

**Comment by dgrammatiko** [comment]
> so the initial view is like this:

Where's the editor in that view?

**Comment by Scrabble96** [comment]
![tinymce-2](https://user-images.githubusercontent.com/36987494/148531893-e02bab25-1436-4c54-8479-324f69873204.jpg)

**Comment by dgrammatiko** [comment]
Go to http://skin.tiny.cloud/t5 and create a custom skin
Then place it in your template CSS folder (IIRC)

**Comment by Scrabble96** [comment]
> Go to http://skin.tiny.cloud/t5 and create a custom skin Then place it in your template CSS folder (IIRC)

Thank you @dgrammatiko - I'll give that a try.

**Comment by brianteeman** [comment]
NO!!!

You're talking at cross purposes
### **This is NOTHING to do with tinymce!!** 

This is not tinymce layout!!!

Your issue is to do with the styling of the modals for the xtd-editor plugins - to confirm that just disable all the xtd-editor plugins and they will disappear

**Comment by Scrabble96** [comment]
Ok. ok. Perhaps the xtd-editor modals plugins CSS could be added to System CSS then?

**Comment by dgrammatiko** [comment]
You can style the XTD by adding CSS to the `layouts/joomla/editors/buttons/button.php` and `layouts/joomla/editors/buttons/modals.php`.

**Comment by brianteeman** [comment]
> This is because my template didn't include 'modal' CSS as I don't use modals anywhere, so the initial view is like this:

If you are using frontend editing and the xtd-editor plugins then you ARE using modals
